be/src/vec/olap/block_reader.cpp

```diff
@@ -129,7 +123,7 @@ OLAPStatus BlockReader::init(const ReaderParams& read_params) {
         for (int j = 0; j < read_params.return_columns.size(); ++j) {
             if (read_params.return_columns[j] == cid) {
                 if (j < _tablet->num_key_columns() || _tablet->keys_type() != AGG_KEYS) {
-                    _normal_columns_idx.emplace_back(j);
+                    _normal_columns.emplace_back(i);
```

Comment:
> `_normal_columns.emplace_back(i);` 这里原来是j，是不是手误写成i了？

no, refer the Proposed changes No.1

Reply:
Here should use the `j` not `i`. because the `origin_return_columns` may different with the `return_columns`. here need a map to insert right data in `insert_data_normal`

Reply:
before modifying: _normal_columns_idx's elements store j in _init_agg_state() function, then used as the _return_columns_loc's subscript. 
express in code: 
_normal_columns_idx[x] = j  and  _return_columns_loc[j] = i 
==>
_return_columns_loc[_normal_columns_idx[x]] equal to i

why not _normal_columns_idx[x] save i directly? there is no need to got i by _return_columns_loc.